Adjust tests for petsc 3.8.3

fdkong opened the pull request:
Closes #11118

fdkong committed:
Make all_print_pps more robust

Closes #11118

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/170371/) on 62dc5f6 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11121/site)

This comment will be updated on new commits.

moosebuild commented:
Job [App tests SLEPc](https://civet.inl.gov/job/170383/) on 62dc5f6 : invalidated by @fdkong

test:OxideEnergyDeposition.matpro_comp

fdkong commented:
@bwspenc  Could you give me insight on the failed BISON tests ?

fdkong committed:
Make TestHarness to take logic "&& ||" for {petsc,slepc}_version

We can do something like petsc_version = '>=3.5.0&&<3.8.0'

Refs #11118

fdkong committed:
New subdomain_partitioner test for petsc-3.8.x

Refs #11118

fdkong committed:
Fix potential race condition for csv_reader.read

vectorpostprocessors/csv_reader.read: Running CSVDiffer.py
vectorpostprocessors/csv_reader.read: In read_out_reader_0000.csv: Header 'month' is missing
vectorpostprocessors/csv_reader.read: In read_out_reader_0000.csv: Header 'day' is missing
vectorpostprocessors/csv_reader.read: In read_out_reader_0000.csv: Header 'year' is missing
vectorpostprocessors/csv_reader.read:
vectorpostprocessors/csv_reader.read: ################################################################################
vectorpostprocessors/csv_reader.read: Tester failed, reason: CSVDIFF
vectorpostprocessors/csv_reader.read:

Refs #11118

fdkong committed:
Try different preconditioner for lots_of_variables.i

Refs #11118

moosebuild commented:
Job [App tests](https://civet.inl.gov/job/172465/) on 67da8bd : invalidated by @fdkong

yak:iqs/1d_pc.diffusion_iqspc_param_adapt:

moosebuild commented:
Job [Modules debug](https://civet.inl.gov/job/172462/) on 67da8bd : invalidated by @fdkong

timeout

fdkong committed:
Loose abs_zero Postprocessor_dt

Rel diff: -4.9005288e-13 ~  1.0458154e-10 = 1.00469e+00

Refs #11118

permcody commented on python/TestHarness/util.py:
```diff
@@ -354,31 +354,71 @@ def getSlepcVersion(libmesh_dir):
+        if len(splits) == 1:
+            return checkLogicVersionSingle(checks, version, package)
+        elif len(splits) == 2:
+            return checkLogicVersionSplits(checks, splits, logic_and, package)
```
This all works, but isn't the cleanest way to do things. "re" can be used to do the split and give back the captured token. I think all of this logic could be compressed into a few lines (maybe 10)... I guess it's not worth really cleaning it up, but keep in mind this is Python, not C.

permcody commented on test/tests/preconditioners/pbp/lots_of_variables.i:
```diff
@@ -9,7 +9,7 @@
     type = PBP
     solve_order = 'vars0 vars1 vars2 vars3 vars4 vars5 vars6 vars7 vars8 vars9 vars10 vars11 vars12 vars13 vars14 vars15 vars16 vars17 vars18 vars19 vars20 vars21 vars22 vars23 vars24 vars25 vars26 vars27 vars28 vars29 vars30 vars31 vars32 vars33 vars34 vars35 vars36 vars37 vars38 vars39 vars40 vars41 vars42 vars43 vars44 vars45 vars46 vars47 vars48 vars49 vars50 vars51 vars52 vars53 vars54 vars5 ...
-    preconditioner = 'AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG AMG A ...
+    preconditioner = 'ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU ILU I ...
```
Changing the preconditioner? Did you talk to @friedmud about this?

permcody commented on test/tests/vectorpostprocessors/csv_reader/tests:
```diff
@@ -16,7 +16,7 @@
     type = RunException
     input = read.i
-    cli_args = "UserObjects/tester/rank=1 UserObjects/tester/gold='1 2 3'"
+    cli_args = "UserObjects/tester/rank=1 UserObjects/tester/gold='1 2 3' Outputs/csv=false"
```
👍

permcody commented on modules/phase_field/src/postprocessors/FeatureFloodCount.C:
```diff
@@ -1652,6 +1652,8 @@ FeatureFloodCount::FeatureData::merge(FeatureData && rhs)
   // Logical AND here to combine flags (INACTIVE & INACTIVE == INACTIVE, all other combos are CLEAR)
   _status &= rhs._status;
 
+  _intersects_boundary = rhs._intersects_boundary;
```
NO! This is definitely wrong. You are adding a line in the "merge" method. How do you combine data structures for two features, one that claims to intersect the boundary and one that does not. This would set the value in the combined feature to whatever is on the right. 

This is the correct logic:
`_intersects_boundary |= rhs._intersects_boundary;`

fdkong commented:
@bwspenc

moosebuild commented:
Job [App tests PETSc alt](https://civet.inl.gov/job/172766/) on 835cb95 : invalidated by @fdkong

Rattlesnake

fdkong commented:
Bison failed because of the changes of METIS. I will test other bison heavy tests to check if or not they are also affected by the changes.

moosebuild commented:
Job [App tests PETSc alt](https://civet.inl.gov/job/172766/) on 835cb95 : invalidated by @fdkong

jwpeterson committed:
Add comments in the test spec file explaining the different restrictions.

jwpeterson commented:
IMO this PR should be merged even if some of the apps don't pass with PETSc-alt yet, since these changes will be required regardless. PETSc 3.9 is also already out, so before we go fixing all the apps for 3.8, we may just be able to skip ahead to that version...

permcody commented:
Sounds fine. I'm approving pending the passage of tests. Anyone can merge once it's done testing.

jwpeterson commented:
@fdkong, why wait? This is complete enough to be merged now. If you have fixes for the apps we can merge them later.

moosebuild commented:
Job [App tests PETSc alt](https://civet.inl.gov/job/177728/) on e14263b : invalidated by @fdkong

Bison

moosebuild commented:
Job [App tests PETSc alt](https://civet.inl.gov/job/177728/) on e14263b : invalidated by @fdkong

Rattlesnake

jwpeterson commented:
@fdkong Do we actually expect the apps to pass with PETSc alt?? If you aren't changing anything on the app side, why keep invalidating?

fdkong commented:
I am actually changing Rattlesnake and Bison.  I think we could merge.

fdkong commented:
We can verify the changes of  Rattlesnake and Bison later in different PR.
